crates/grafbase-sdk/src/host_io/http.rs

```diff
@@ -333,3 +344,45 @@ impl Default for BatchHttpRequest {
         Self::new()
     }
 }
+
+/// A struct that represents an HTTP response.
+pub struct HttpResponse(crate::wit::HttpResponse);
+
+impl HttpResponse {
+    /// Returns the status code of the HTTP response.
+    pub fn status(&self) -> http::StatusCode {
+        *OnceCell::new()
+            .get_or_init(|| http::StatusCode::from_u16(self.0.status).expect("must be valid, this comes from reqwest"))
+    }
+
+    /// Returns the headers of the HTTP response.
+    pub fn headers(&self) -> &[(String, String)] {
+        &self.0.headers
```

Comment:
For ergonomics, we could return an `http::HeaderMap` here, but it's fine for the stage we're at now.

Reply:
Yeah, and no right now. It's annoying to build and let's see if we need it.